eclair-core/src/main/scala/fr/acinq/eclair/payment/Relayer.scala

```diff
@@ -66,27 +69,35 @@ class Relayer(nodeParams: NodeParams, register: ActorRef, paymentHandler: ActorR
 
   private val commandBuffer = context.actorOf(Props(new CommandBuffer(nodeParams, register)))
 
-  override def receive: Receive = main(Map.empty, new mutable.HashMap[PublicKey, mutable.Set[ShortChannelId]] with mutable.MultiMap[PublicKey, ShortChannelId])
+  override def receive: Receive = main(Map.empty, new mutable.HashMap[PublicKey, mutable.Set[ShortChannelId]] with mutable.MultiMap[PublicKey, ShortChannelId], Map.empty)
 
-  def main(channelUpdates: Map[ShortChannelId, OutgoingChannel], node2channels: mutable.HashMap[PublicKey, mutable.Set[ShortChannelId]] with mutable.MultiMap[PublicKey, ShortChannelId]): Receive = {
+  def main(channelUpdates: Map[ShortChannelId, OutgoingChannel],
```

Comment:
I'd rather redefine `OutgoingChannel` as:

```scala
case class OutgoingChannel(nextNodeId: PublicKey, channelUpdate: ChannelUpdate, commitments: Commitments)
```
Change is minimal, and then you can implement `GetUsableBalances` in a self-contained way by iterating over the `channelUpdates` map.